### PR TITLE
draft of telegraf

### PR DESCRIPTION
**Pull request by kelwang**
Closes #https://github.com/influxdata/platform/issues/946
#https://github.com/influxdata/platform/issues/1010

 - [x] Rebased/mergeable

### CODE REVIEW

File: http/swagger.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ servers:
   - url: /api/v2
 paths:
   /signin:
-    servers:
```
Comment (goller) [inline http/swagger.yml]: the servers section should be removed all together.

File: http/swagger.yml [hunk 2]
```diff
@@ -74,6 +74,289 @@ paths:
...
+      summary: Create a telegraf config
+      parameters:
+          - in: query
+            name: org
+            description: specifies the organization of the resource
+            required: true
+            schema:
+              type: string
+      requestBody:
+        description: bucket to create
```
Comment (goller) [inline http/swagger.yml]: telegraf config

File: http/swagger.yml [hunk 3]
```diff
@@ -74,6 +74,289 @@ paths:
...
+      requestBody:
+        description: user to add as member
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/User"
+      responses:
+        '201':
+          description: member added to bucket
```
Comment (goller) [inline http/swagger.yml]: member added to telegraf

File: http/swagger.yml [hunk 4]
```diff
@@ -74,6 +74,289 @@ paths:
...
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+  '/telegrafs/{telegrafID}/members/{userID}':
+    delete:
+      tags:
+        - Users
+        - Buckets
```
Comment (goller) [inline http/swagger.yml]: telegraf

File: http/swagger.yml [hunk 5]
```diff
@@ -3818,6 +4101,61 @@ components:
...
+        owners:
+          $ref: "#/components/schemas/Owners"
+        version:
+          type: string
+        agent:
+          type: object
+          properties:
+            interval:
+              type: string
+              example: "13m"
```
Comment (goller) [inline http/swagger.yml]: this is an incorrect example

File: http/swagger.yml [hunk 6]
```diff
@@ -3818,6 +4101,61 @@ components:
...
+        plugins:
+          type: object
+          additionalProperties:
+            type: object
+            properties:
+              code:
```
Comment (goller) [inline http/swagger.yml]: Isn't this type?

File: http/swagger.yml (hunk 5 above)
Comment (kelwang) [reply http/swagger.yml]: ?

File: http/swagger.yml (hunk 6 above)
Comment (kelwang) [reply http/swagger.yml]: It's the map type of swagger, since we won't use map any more, this is removed.

File: http/swagger.yml [hunk 7]
```diff
@@ -4180,6 +4469,52 @@ components:
...
+          type: object
+          properties:
+            name:
+              type: string
+            type:
+              type: string
+              enum: [input, output, processor, aggregator]
+            comment: 
+              type: string
+            configs: {}
```
Comment (goller) [inline http/swagger.yml]: configs needs an "anyOf" of the several kinds of configurations.

File: http/swagger.yml [hunk 8]
```diff
@@ -4180,6 +4469,52 @@ components:
...
+          type: object
+          properties:
+            interval:
+              type: string
+              example: "13m"
+        plugins:
+          type: object
+          properties:
+            name:
+              type: string
```
Comment (goller) [inline http/swagger.yml]: this needs an enum as well for the plugin names.

**Pull request merged**